scrapinghub/client.py

```diff
@@ -0,0 +1,370 @@
+        self._client = client
+        self._origin = cls(client._hsclient, key)
+
+        if issubclass(cls, ItemsResourceType):
+            self._proxy_methods(['get', 'write', 'flush', 'close',
+                                 'stats', ('iter', 'list')])
+
+        # DType iter_values() has more priority than IType list()
+        if issubclass(cls, DownloadableResource):
+            methods = [('iter', 'iter_values'),
+                       ('iter_raw_msgpack', 'iter_msgpack'),
+                       ('iter_raw_json', 'iter_json')]
+            self._proxy_methods(methods)
+
+            # apply_iter_filters is responsible to modify filter params for all
+            # iter* calls: should be used only if defined for a child class
+            if hasattr(self, '_modify_iter_filters'):
+                apply_fn = getattr(self, '_modify_iter_filters')
+                for method in [method[0] for method in methods]:
+                    wrapped = wrap_kwargs(getattr(self, method), apply_fn)
+                    setattr(self, method, wrapped)
+
+    def _proxy_methods(self, methods):
+        proxy_methods(self._origin, self, methods)
+
+
+class Logs(_Proxy):
+
+    def __init__(self, *args, **kwargs):
+        super(Logs, self).__init__(*args, **kwargs)
+        self._proxy_methods(['log', 'debug', 'info', 'warning', 'warn',
+                             'error', 'batch_write_start'])
+
+    def _modify_iter_filters(self, params):
+        offset = params.pop('offset', None)
+        if offset:
+            params['start'] = '{}/{}'.format(self.key, offset)
+        level = params.pop('level', None)
+        if level:
+            minlevel = getattr(LogLevel, level, None)
+            if minlevel is None:
+                raise APIError("Unknown log level: {}".format(level))
+            params['filter'] = json.dumps(['level', '>=', [minlevel]])
+        return params
+
+
+class Items(_Proxy):
+
+    def _modify_iter_filters(self, params):
+        offset = params.pop('offset', None)
+        if offset:
+            params['start'] = '{}/{}'.format(self.key, offset)
+        return params
+
+
+class Requests(_Proxy):
+
+    def __init__(self, *args, **kwargs):
+        super(Requests, self).__init__(*args, **kwargs)
+        self._proxy_methods(['add'])
+
+
+class Samples(_Proxy):
+    pass
+
+
+class Activity(_Proxy):
+
+    def __init__(self, *args, **kwargs):
+        super(Activity, self).__init__(*args, **kwargs)
+        self._proxy_methods([('iter', 'list')])
+
+    def add(self, *args, **kwargs):
+        self._origin.add(*args, **kwargs)
+
+    def post(self, *args, **kwargs):
+        self._origin.post(*args, **kwargs)
+
+
+class Collections(_Proxy):
+
+    def __init__(self, *args, **kwargs):
+        super(Collections, self).__init__(*args, **kwargs)
+        self._proxy_methods([
+            'count', 'get', 'set', 'delete', 'create_writer',
+            '_validate_collection',
+        ])
+
+    def new_store(self, colname):
```

Comment:
IMO `new_` should be replaced with `get_` for all the following methods. it's not new collection if it was already created and `project.collections.get_store('foo')` looks a bit more relevant here.